data-migration/migrate-with-pt-ghost.md

```diff
@@ -0,0 +1,67 @@
+
+When using DM to migrate data from MySQL to TiDB, you can enbale `online-ddl` to allow collaboration of DM and gh-ost or pt-osc.
+
+For the detailed replication instruction, refer to the following documents by scenarios:
+
+- [Migrate MySQL Data of Less Than 1 TiB to TiDB](/data-migration/migrate-mysql-tidb-less-tb.md)
+- [Migrate MySQL Data of More Than 1 TiB to TiDB](/data-migration/migrate-mysql-tidb-above-tb.md)
+- [Migrate and Merge MySQL Shards of Less Than 1 TiB to TiDB](/data-migration/migrate-shared-mysql-tidb-less-tb.md)
+- [Migrate and Merge MySQL Shards of More Than 1 TiB to TiDB](/data-migration/migrate-shared-mysql-tidb-above-tb.md)
+
+## Enable online-ddl on DM
+
+In the task configuration file of DM, set the global parameter `online-ddl` to `true`, as shown below:
+
+```yaml
+# ----------- Global configuration -----------
+## ********* Basic configuration *********
+name: test                      # The name of the task. Should be globally unique.
+task-mode: all                  # The task mode. Can be set to `full`, `incremental`, or `all`.
+shard-mode: "pessimistic"       # The shard merge mode. Optional modes are `pessimistic` and `optimistic`. The `pessimistic` mode is used by default. After understanding the principles and restrictions of the "optimistic" mode, you can set it to the "optimistic" mode.
+meta-schema: "dm_meta"          # The downstream database that stores the `meta` information.
+online-ddl: true                # Enable online-ddl support on DM to support automatic processing of "gh-ost" and "pt" for the upstream database.
+```
+
+## Workflow after enabling online-ddl
+
+After online-ddl is enabled on DM, the DDL statements generated by DM replicating gh-ost or pt-osc will change.
+
+The workflow of gh-ost or pt-osc:
+
+- Create a ghost table according to the table schema of the DDL real table.
+
+- Apply DDLs on the ghost table.
+
+- Replicate the data of the DDL real table to the ghost table.
+
+- After the data are consistent between the two tables, use the rename statement to replace the real table with the ghost table.
+
+The workflow of DM:
+
+- Skip creating the ghost table downstream.
+
+- Record DDLs applied to the ghost table.
+
+- Replicate data only from the ghost table.
+
+- Apply DDLs recorded downstream.
+
+![dm-online-ddl](/media/dm/dm-online-ddl.png)
+
+The change in the workflow brings the following advantages:
+
+- The downstream TiDB does not need to create and replicate the ghost table, saving the storage space and network transmission overhead.
+
+- When you migrate and merge data from sharded tables, the RENAME operation is ignored for each sharded ghost table to ensure the correctness of the replication.
+
```

Comment:
Some documents are missing:
```
如果您想深入了解其实现原理，请阅读以下两篇技术博客：

- [DM 源码阅读系列文章（八）Online Schema Change 迁移支持](https://pingcap.com/blog-cn/dm-source-code-reading-8/#dm-源码阅读系列文章八online-schema-change-迁移支持)
- [TiDB Online Schema Change 原理](https://pingcap.com/zh/blog/tidb-source-code-reading-17)
```

Reply:
> Some documents are missing:
> 
> ```
> 如果您想深入了解其实现原理，请阅读以下两篇技术博客：
> 
> - [DM 源码阅读系列文章（八）Online Schema Change 迁移支持](https://pingcap.com/blog-cn/dm-source-code-reading-8/#dm-源码阅读系列文章八online-schema-change-迁移支持)
> - [TiDB Online Schema Change 原理](https://pingcap.com/zh/blog/tidb-source-code-reading-17)
> ```

Sharp eyes!
We have no English version of these blogs and I have confirmed with Zhaoyang about it. He suggested we keep the links in Chinese but remove them from the English version. Do you have any other suggestions?

Reply:
> He suggested we keep the links in Chinese but remove them from the English version.
SGTM